cartofacade/cgo_api.go

```diff
@@ -0,0 +1,400 @@
+
+	#include "../viam-cartographer/src/carto_facade/carto_facade.h"
+
+	viam_carto_lib* alloc_viam_carto_lib() { return (viam_carto_lib*) malloc(sizeof(viam_carto_lib)); }
+	void free_alloc_viam_carto_lib(viam_carto_lib* p) { free(p); }
+
+	viam_carto* alloc_viam_carto() { return (viam_carto*) malloc(sizeof(viam_carto)); }
+	void free_alloc_viam_carto(viam_carto* p) { free(p); }
+
+	bstring* alloc_bstring_array(size_t len) { return (bstring*) malloc(len * sizeof(bstring)); }
+	void free_bstring_array(bstring* p) { free(p); }
```

Comment:
@nicksanford I just remembered you mentioned that we might need to handle a case where C could not allocate memory here. Do you have any suggestions on how best to do it?

Reply:
Yes. if `alloc_viam_carto` returns `nullptr` then it means that allocation has failed & the go function which called it should return early with an error.